### PR TITLE
Describe performance saturation experiments

### PR DESCRIPTION
**Pull request by sodar**

### CODE REVIEW

File: doc/For developers/Developer guide/Performance/Testing.md [hunk 1]
```diff
@@ -329,4 +328,75 @@ sender   > .\NTttcp.exe -s -m 1,*,172.16.0.22 -l 128k -t 15
...
+### Raw vs Hyper-V performance
+
+According to the Microsoft Networking blog (here: [VMQ Deep Dive][vmq]) the following behavior can be observed:
+
+- Without any VMSwitch/Hyper-V configured, packets are directed to different hardware queues based on flow hash.
+  Each queue is assigned to a different CPU core.
+  Windows utilizes RSS on NICs to achieve that.
+- With VMSwitch configured, packets are directed to different hardware queues based on destination MAC address.
+  Each queue is assigned to a different CPU core.
+  As a result each virtual adapter is assigned to a single CPU core and network throughput should be capped at c. 2-3 GBps.
```
Comment (m-kostrzewa) [inline doc/For developers/Developer guide/Performance/Testing.md]: 1. Shouldn't it be Gbps, not GBps?
2. This is assuming 4 cores right? And it's exactly around 2.5 then?

File: doc/For developers/Developer guide/Performance/Testing.md [hunk 2]
```diff
@@ -329,4 +328,75 @@ sender   > .\NTttcp.exe -s -m 1,*,172.16.0.22 -l 128k -t 15
...
+| ID  | Test name               | Description |
+|-----|-------------------------|-------------|
+| A   | Raw                     | Raw Windows Server network stack |
+| B   | 2 Containers            | 2 WinSrv containers on 2 compute nodes (1 container per node) |
+| C   | 4 Containers            | 4 WinSrv containers on 2 compute nodes (2 containers per node) |
+| D   | 2 Containers, 2 threads | Same as B, but NTttcp uses 2 threads instead of one |
+
+If assumption is correct, then:
+
+- network throughput in scenario `C` should be twice as high as in `B`,
```
Comment (m-kostrzewa) [inline doc/For developers/Developer guide/Performance/Testing.md]: We mean _total_ throughput, right? If so, let's clarify it here

File: doc/For developers/Developer guide/Performance/Testing.md [hunk 3]
```diff
@@ -40,8 +40,7 @@ Conclusions:
 
 - Enabling Hyper-V on Windows Server 2016 reduces TCP throughput by a factor of 3-4.
```
Comment (jacekchalupka) [inline doc/For developers/Developer guide/Performance/Testing.md]: Add information which exactly test cases support this conclusions (for this one A and B test case)

**Review by jacekchalupka — CHANGES_REQUESTED** [review]

File: doc/For developers/Developer guide/Performance/Testing.md (hunk 1 above)
Comment (sodar) [reply doc/For developers/Developer guide/Performance/Testing.md]: 1. Yes, it should be Gbps.
2. 2-3 Gbps is a number achievable by single core, no matter how many cores are in the system.

**Review by jacekchalupka — APPROVED** [review]

**Pull request merged**